src/schema/v2/me/dislikeArtworkMutation.ts

```diff
@@ -0,0 +1,41 @@
+import { GraphQLBoolean, GraphQLNonNull, GraphQLString } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import { ArtworkType } from "../artwork"
+
+export default mutationWithClientMutationId<any, any, ResolverContext>({
+  name: "DislikeArtwork",
+  description: "Add (or remove) an artwork to (from) a users dislikes.",
+  inputFields: {
+    artworkID: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    remove: {
+      type: new GraphQLNonNull(GraphQLBoolean),
+    },
+  },
+  outputFields: {
+    artwork: {
+      type: ArtworkType,
+      resolve: ({ artwork_id }, _, { artworkLoader }) =>
+        artworkLoader(artwork_id),
+    },
+  },
+  mutateAndGetPayload: async (
+    { artworkID: artwork_id, remove },
+    { userID, dislikeArtworkLoader, deleteDislikedArtworkLoader }
+  ) => {
+    if (!deleteDislikedArtworkLoader || !dislikeArtworkLoader) {
+      return new Error("You need to be signed in to perform this action")
+    }
+
+    const loader = remove ? deleteDislikedArtworkLoader : dislikeArtworkLoader
+    try {
+      return loader(artwork_id, { user_id: userID }).then(() => ({
+        artwork_id,
+      }))
```

Comment:
So this try and catch doesn't do anything without awaiting on the loader. This actually would never throw (unless something internal to the loader thru, not the actual promise rejection)